### PR TITLE
Fix format of advanced.rst according to PEP-8 standard

### PR DESCRIPTION
**Pull request by YuechengWu**
- [x] closes #23790 
- [ ] tests added / passed
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

**Review by datapythonista — COMMENTED** [review]
looks great, added few comments

File: doc/source/advanced.rst [hunk 1]
```diff
@@ -328,17 +330,18 @@ As usual, **both sides** of the slicers are included as this is label indexing.
...
+   miindex = pd.MultiIndex.from_product([mklbl('A', 4),
+                                         mklbl('B', 2),
+                                         mklbl('C', 4),
+                                         mklbl('D', 2)])
+   micolumns = pd.MultiIndex.from_tuples([('a', 'foo'), ('a', 'bar'),
+                                          ('b', 'foo'), ('b', 'bah')],
                                          names=['lvl0', 'lvl1'])
-   dfmi = pd.DataFrame(np.arange(len(miindex)*len(micolumns)).reshape((len(miindex),len(micolumns))),
+   dfmi = pd.DataFrame(np.arange(len(miindex) * len(micolumns))
+                       .reshape((len(miindex), len(micolumns))),
```
[the remark this answers is not in the document]
Comment (YuechengWu) [reply doc/source/advanced.rst]: removed comma

File: doc/source/advanced.rst [hunk 2]
```diff
@@ -235,7 +237,7 @@ keys take the form of tuples. For example, the following works as you would expe
 
    df = df.T
    df
-   df.loc[('bar', 'two'),]
+   df.loc[('bar', 'two'), ]
```
[the remark this answers is not in the document]
Comment (YuechengWu) [reply doc/source/advanced.rst]: removed comma

File: doc/source/advanced.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (YuechengWu) [reply doc/source/advanced.rst]: properly aligned

File: doc/source/advanced.rst [hunk 3]
```diff
@@ -688,12 +691,12 @@ faster than fancy indexing.
    indexer = np.arange(10000)
    random.shuffle(indexer)
 
-   timeit arr[indexer]
-   timeit arr.take(indexer, axis=0)
+   timeit(arr[indexer])
+   timeit(arr.take(indexer, axis=0))
```
[the remark this answers is not in the document]
Comment (YuechengWu) [reply doc/source/advanced.rst]: it still gives me a syntax error when i use %timeit

**Review by datapythonista — APPROVED** [review]
lgtm, thank @YuechengWu 

The CI is green, if it doesn't work locally I guess it's because you miss some ipython dependency.

**Comment by YuechengWu** [comment]
cool, thanks @datapythonista

**Comment by YuechengWu** [comment]
@gfyoung My pleasure!

**Comment by datapythonista** [comment]
@gfyoung we have `flake8-rst` activated, but the version we have doesn't check `ipython` blocks. We're upgrading the version in #23847, but we still have plenty of files to fix.

**Comment by gfyoung** [comment]
@datapythonista : Ah, right.  Thanks for clarifying!

**Pull request merged**